openshift/oadp-operator
# feat: Integrate Non-Admin Controller (NAC) with OADP

## Why the changes were made

Add Non-Admin Controller (NAC) to OADP to start testing it. Fixes https://github.com/migtools/oadp-non-admin/issues/19

## How the changes were made

Studied how to "reverse engineered" this PR https://github.com/openshift/oadp-operator/pull/1171

Then applied necessary changes to `migtools/oadp-non-admin` repo

Created command to update non admin controller Kubernetes objects

## How to test the changes made

Read https://github.com/migtools/oadp-non-admin/pull/16

To test update command, change anything in newly added files in `config/` folder and run
```sh
git clone --depth=1 git@github.com:mateusoliveira43/oadp-non-admin -b fix/contribution-doc
NON_ADMIN_CONTROLLER_PATH=<path_to_git_clone> make update-non-admin-manifests
```
Changes should be reverted

Also, check if all necessary files from NAC were included here.

Run `make deploy-olm` and create DPA with new feature
```yaml
...
spec:
  features:
    nonAdmin:
      enable: true
...
```

You should be able to:
- deploy Non-Admin Controller (`spec.features.nonAdmin.enable` true)
- undeploy Non-Admin Controller (`spec.features.nonAdmin.enable` false or unset)
- use custom Non-Admin Controller image using environment variable and unsupported override
  To create custom image, run
  ```sh
  git clone --depth=1 git@github.com:mateusoliveira43/oadp-non-admin -b fix/contribution-doc
  IMG=ttl.sh/<unique_name_from_your_head>:1h make docker-build docker-push
  ```
